dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -461,47 +371,64 @@ def find_counterfactuals(self, query_instance, desired_range, desired_class, fea
         iterations = 0
         previous_best_loss = -np.inf
         current_best_loss = np.inf
-        current_best_cf = []
         stop_cnt = 0
         cfs_preds = [np.inf]*self.total_CFs
-        while iterations < maxiterations:
-            if abs(previous_best_loss - current_best_loss) <= 1e-2: #and (self.model.model_type == 'classifier' and all(i == desired_class for i in cfs_preds) or (self.model.model_type == 'regressor' and all(desired_range[0] <= i <= desired_range[1] for i in cfs_preds))):
+        to_pred = None
+
+        while iterations < maxiterations or len(population) == self.total_CFs:
+            start = timeit.default_timer()
+            if abs(previous_best_loss - current_best_loss) <= thresh and (self.model.model_type == 'classifier' and all(i == desired_class for i in cfs_preds) or (self.model.model_type == 'regressor' and all(desired_range[0] <= i <= desired_range[1] for i in cfs_preds))):
```

Comment:
the second two conditions can be checked by simply invoking `decide_cf_validity` from explainer_base.